sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,575 @@
+        :type headers: HttpResponse Headers
+        :param table_entity_properties: The properties for the table entity.
+        :type table_entity_properties: dict[str, object]
+        :param query_options: Parameter group.
+        :type query_options: ~azure.table.models.QueryOptions
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: dict mapping str to object, or the result of cls(response)
+        :rtype: dict[str, object] or None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if table_entity_properties:
+
+            if "PartitionKey" in table_entity_properties and "RowKey" in table_entity_properties:
+                table_entity_properties = _add_entity_properties(table_entity_properties)
+            else:
+                raise ValueError
+        try:
+
+            inserted_entity = self._client.table.insert_entity(
+                table=self.table_name,
+                table_entity_properties=table_entity_properties,
+                query_options=query_options,
+                **dict(kwargs, headers=headers)
+            )
+            properties = _convert_to_entity(inserted_entity)
+            return Entity(properties)
+        except ValueError as error:
+            process_storage_error(error)
+
+    @distributed_trace
+    def update_entity(
+            self,
+            partition_key=None,
+            row_key=None,
+            etag=None,
+            match_condition=None,
+            response_hook=None,  # pylint:disable=W0613
+            table_entity_properties=None,
+            **kwargs
+    ):
+        # type: (...) -> None
+        """Update entity in a table.
+
+        :param response_hook:
+        :param table_entity_properties: The properties for the table entity.
+        :type table_entity_properties: dict[str, object]
+        :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :param etag: Etag of the entity
+        :type etag: str
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: None, or the result of cls(response)
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        if table_entity_properties:
+            partition_key = table_entity_properties['PartitionKey']
+            row_key = table_entity_properties['RowKey']
+            table_entity_properties = _add_entity_properties(table_entity_properties)
```

Comment:
above we threw a ValueError if the partition key and row key weren't present. do we need to do the same here?
Edit: also for merge_entity